### PR TITLE
Sending receipt for account-payed bookings only when actual charge occurs

### PR DESCRIPTION
**Pull request by Raibo**

### CODE REVIEW

**Review by kirillta — APPROVED** [review]

File: Api/Services/Accommodations/Bookings/BookingDocumentsService.cs [hunk 1]
```diff
@@ -148,9 +149,17 @@ static BookingInvoiceData.SellerInfo GetSellerDetails(Data.Booking.Booking booki
                 counterparty.Phone,
                 counterparty.BillingEmail);
         }
-        
 
-        public async Task<Result<(DocumentRegistrationInfo RegistrationInfo, PaymentReceipt Data)>> GenerateReceipt(int bookingId, AgentContext agent)
+
+        public Task<Result<(DocumentRegistrationInfo RegistrationInfo, PaymentReceipt Data)>> GenerateReceipt(int bookingId, AgentContext agent) =>
```
[the remark this answers is not in the document]
Comment (Raibo) [reply Api/Services/Accommodations/Bookings/BookingDocumentsService.cs:154]: We don't have `AgentContext` when calling `Charge` from `BookingsProcessingService`, so have to use `Agent` instead.

File: Api/Services/Accommodations/Bookings/BookingPaymentService.cs [hunk 2]
```diff
@@ -53,25 +58,53 @@ public async Task<Result<string>> Capture(Booking booking, UserInfo user)
...
         public async Task<Result<string>> Charge(Booking booking, UserInfo user)
         {
-            if (booking.PaymentMethod != PaymentMethods.BankTransfer)
+            return await EnsureBankTransfer()
+                .Bind(Charge)
+                .Bind(SendReceipt)
+                .Finally(WriteLog);
+
+
+            Result<string> EnsureBankTransfer() =>
```
[the remark this answers is not in the document]
Comment (Raibo) [reply Api/Services/Accommodations/Bookings/BookingPaymentService.cs:67]: I initially thought that we have to maintain Result<string> type, but there is an implicit conversion (Result -> Result<T>) when the result is failure.
Fixed.

[the remark this answers is not in the document]
Comment (Raibo) [reply Api/Services/Accommodations/Bookings/BookingPaymentService.cs:67]: Renamed to `CheckPaymentMethod`

File: Api/Services/Accommodations/Bookings/BookingDocumentsService.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Raibo) [reply Api/Services/Accommodations/Bookings/BookingDocumentsService.cs:154]: Also, I don't want `GenerateReceipt` to be callable with just strings of first and second name from outside of the class.

**Review by oleg1std — APPROVED** [review]

**Review by mingazhev — APPROVED** [review]

**Pull request merged**